template/node/src/service.rs

```diff
@@ -444,7 +470,7 @@ pub fn start_instant_seal_node(config: Configuration) -> Result<TaskManager, sc_
 				transaction_pool: Some(OffchainTransactionPoolFactory::new(
 					transaction_pool.clone(),
 				)),
-				network_provider: network.clone(),
+				network_provider: Arc::new(network.clone()),
```

Comment:
This looks wrong but I see polkadot-sdk also does it...

Reply:
yeah the compiler did not complaint about it, why do you think is wrong?

Reply:
If `network` is already an `Arc`, then this is like having an `Arc<Arc<T>>`, it works but it doesn't make sense.